aws/tfstate-backend/install.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+## Spaces before and after `backend` required to select right word, because `backend` appears 3 times in main.tf
+sed -i "s/ backend / #backend /" main.tf
+sed -i "s/ role_arn / #role_arn /" main.tf
```

Comment:
This can't be generally true. This is maybe true for the "root" org, but certainly not on subaccounts.

Reply:
roles are how we determine the account in which to provision the resources, so if this is not set, terraform won't use the appropriate account.

Reply:
I think this will need to be an arg. e.g. `--disable-role-arn`

Reply:
yes, we disable the role only in the `root` account and only once at cold start when we provision `iam` and we don't have any roles yet

Reply:
So the way I typically do something like this, is:
have two different `make` targets with
two different "invisible-to-terraform™" versions of the resource definitions:
- `provider-with-assume-role.tf.in`
- `provider-bootstrap.tf.in`

and the `make` targets look like:
```
bootstrap:
    rm -f provider.tf
    ln -s provider-bootstrap.tf.in provider.tf

default:
   rm -f provider.tf
   ln -s provider-with-assume-role.tf.in provider.tf
```

and the install script then looks closer to:
```
init-terraform
make bootstrap
terraform plan -input=false -out=install-plan &&
  terraform apply -input=false install-plan &&
  mv -v install-plan install-plan.applied
```